### ISSUE TITLE
I have been experiencing more frequent crashing of Explorer

### ISSUE THREAD

**Issue by bonzibudd** [Open-Shell/Open-Shell-Menu#490]
**Describe the bug**
As of recently, I have begun experiencing Explorer crashes which I can only attribute to installing Open-Shell. The behavior seems to suggest this as well. Sometimes when I begin searching, Open-Shell freezes and spontaneously crashes Explorer. Sometimes this causes a “Crash loop,” but most of the time it just crashes and restarts. I am thinking that this has something to do with an error in the index, but I have no real way of knowing.

**To Reproduce**
Steps to reproduce the behavior (Only happens occasionally):
1. Open the Start menu
2. Begin a search
3. Explorer will hang after typing a few letters, and quickly crash

**Expected behavior**
That shouldn’t happen

**Video/Technical info**
I am not at a Windows system to test this right now, however I will try my best to get a video and/or a “Trace” log. It may be difficult since it is seemingly random when it occurs, and it may also change depending on how many other processes are running. I could also provide a screenshot of the Event Viewer log which appears. What I can say is that the Event log shows the faulting application as “Explorer.exe” and gives the keyword “Classic.”

**Version:**
 - Open-Shell: Latest nightly
 - OS: Win10 ver. 2004

**Additional context**
I would say the frequency of this has been every 5 days or so. It probably increases the more I use the system. While most of the time this bug completely crashes Explorer, I have seen instances of just a hang, and then Explorer returns. I would say it is also possible that the fix for launching the Settings index with no arguments could also be at play here. I can’t say for certain if that issue has occurred since I installed that fix, but I believe that it has.

The only reason I have to assume this is an Open-Shell issue is because I have had it occur on a freshly installed VM with nothing but a browser and Open-Shell installed. I got a very similar crash to what I was experiencing on my main system, which is where I first noticed the issue.

Additionally, I have been experiencing a completely random Explorer crash, however that one I can’t attribute to Open-Shell. I’ve only experienced it on my main system, and it might as well be any other of the dozens of programs I have installed there. Because of this, I won’t make an issue about this yet.

**Comment by ge0rdi** [comment]
Please provide crash dump obtained using this [guide](https://github.com/Open-Shell/Open-Shell-Menu/wiki/How-to-create-dump-on-Windows-Explorer-crash).

**Comment by bonzibudd** [comment]
@ge0rdi Sorry for the delay, I'm working on it. Unfortunately I got the crash right as I was typing in a search to open the command, lol.

**Comment by bonzibudd** [comment]
I think I finally got it: [OpenShellDMP](https://1drv.ms/u/s!AqxEMH2XuF6DxQdrKgwpYrmZkBE-?e=Hpzdwl)

**Comment by bonzibudd** [comment]
I think I have found what might be causing it. It seems to happen when you click "Reset all settings" or "Clear cached information." I'm not sure if this is the only instance when it happens, though.

**Comment by blackcrack** [comment]
> 
> 
> Please provide crash dump obtained using this [guide](https://github.com/Open-Shell/Open-Shell-Menu/wiki/How-to-create-dump-on-Windows-Explorer-crash).

**Comment by ge0rdi** [comment]
@bonzibudd 
Thanks for the dump.
I can confirm there is race-condition when using `Clear cached information`. Search thread uses data that were already freed.

Actually I have this encountered few times already and have it on my TODO list.
But since clearing cached info isn't used very often (besides of development), it din't have much priority.
I'll try to have a look eventually.

For now when I'm clearing caches (during development), I'm also exiting `Open-Shell` and then start  _fresh_ instance.

**Comment by bonzibudd** [comment]
Ok, thanks for the confirmation. I guess it's good that it only affects the "Clear cached information" setting.

**Comment by LinuxOnTheDesktop** [comment]
I just installed the latest release (atop my existing version) and generated what looks like an OpenShell crash and restart - as follows. Having installed the latest version, I went to the OpenShell search settings and enabled both autocomplete and frequency trapping. Then I did a search. Then the search box disappeared and the hourglass appeared. Clicking Start product up the search again, and this time it worked.

**Comment by ge0rdi** [comment]
@LinuxOnTheDesktop 
If you are experiencing crashes and can replicate it, please, provide crash dump obtained using this [guide](https://github.com/Open-Shell/Open-Shell-Menu/wiki/How-to-create-dump-on-Windows-Explorer-crash).
Without dump it will be pretty much impossible to tell what's going on.

**Comment by LinuxOnTheDesktop** [comment]
I had trouble getting ProcDump working properly: it kept complaining of multiple Explorer processes. But I closed all but one and then ran ProcDump. After that, I was unable to reproduce the crash, even when I toggled the status of some search options. Perhaps the problem I had owed to upgrading OpenShell - and now the upgrade is done, and so there is no crash. At any rate, I've decided that it is easier to use OpenShell for the menu for to use Windows's own search box for searching.
